elephant/conversion.py

```diff
@@ -152,3 +155,704 @@ def binarize(spiketrain, sampling_rate=None, t_start=None, t_stop=None,
+        """
+        Checks given parameter.
+        Calculates also missing parameter.
+
+        Parameters
+        ----------
+        binsize : quantity.Quantity
+            Size of Bins
+        num_bins : int
+            Number of Bins
+        t_start: quantity.Quantity
+            Start time of the spike
+        t_stop: quantity.Quantity
+            Stop time of the spike
+
+        Raises
+        ------
+        ValueError :
+            If all parameters are `None`, a ValueError is raised.
+        TypeError:
+            If type of :attr:`num_bins` is not an Integer.
+        """
+        # Raise error if no argument is given
+        if binsize is None and t_start is None and t_stop is None \
+                and num_bins is None:
+            raise ValueError(
+                "No arguments given. Please enter at least three arguments")
+        # Check if num_bins is an integer (special case)
+        if num_bins is not None:
+            if type(num_bins) is not int:
+                raise TypeError("num_bins is not an integer!")
+        # Check if all parameters can be calculated, otherwise raise ValueError
+        if t_start is None:
+            self.t_start = calc_tstart(num_bins, binsize, t_stop)
+        elif t_stop is None:
+            self.t_stop = calc_tstop(num_bins, binsize, t_start)
+        elif num_bins is None:
+            self.num_bins = calc_num_bins(binsize, t_start, t_stop)
+            if self.matrix_columns is None:
+                self.matrix_columns = self.num_bins
+        elif binsize is None:
+            self.binsize = calc_binsize(num_bins, t_start, t_stop)
+
+    def __calc_start_stop(self, spiketrains):
+        """
+        Calculates start, stop from given spike trains.
+
+         The start and stop points are calculated from given spike trains, only
+         if they are not calculable from given parameter or the number of
+         parameters is less than three.
+
+        """
+        if self.__count_params() is False:
+            if self.t_stop is None:
+                start, stop = set_start_stop_from_input(spiketrains)
+                self.t_start = start
+            if self.t_stop is None:
+                if stop:
+                    self.t_stop = stop
+                else:
+                    self.t_stop = set_start_stop_from_input(spiketrains)[1]
+
+    def __count_params(self):
+        """
+        Counts the parameter and returns **True** if the count is greater
+        or equal to `3`.
+
+        The calculation of the binned matrix is only possible if there are at
+        least three parameter (fourth parameter will be calculated out of
+        them).
+        This method counts the necessary parameter and returns **True** if the
+        count is greater or equal to `3`.
+
+        Returns
+        -------
+        bool :
+            True, if the count is greater or equal to `3`.
+            False, otherwise.
+
+        """
+        param_count = 0
```

Comment:
No, the above is easier ;) 
Do you mean `Zero` instead of `None`? If `t_start` and `t_stop` are both zero the test later will fail. Also with `None` it will fail later on.